### DISCUSSION TITLE
Cruiser 3.0.32 (Beta)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#299] [Announcements]
Cruiser 3.0.32 (Beta) is available:

https://www.dropbox.com/sh/itq3e3d0wr0zp3q/AADZ0BYypUf91GhMVKryZzmba

- https://github.com/devemux86/cruiser/discussions/165#discussioncomment-6800089

- https://github.com/devemux86/cruiser/discussions/298

- https://github.com/devemux86/cruiser/discussions/292

_**Please backup your favorites, routes, settings before testing Beta versions.**_

<img src="https://github.com/devemux86/cruiser/assets/3484020/16d50060-2ae6-47b0-bf07-86c8c9d38de1" width="50%"/>

**Comment by Ciutnik** [comment]
Thanks for the new version! 👍I agree that it is necessary to make a backup of the preferences, because they reset to default.

As for the translation (to Polish), I already found a few weird misses and one typo (shame!) 😱

**Reply by devemux86** [reply]
> it is necessary to make a backup of the preferences, because they reset to default.

What do you mean with the "reset to default", when this happen?

It's a **Beta** version, so a separate apk, for users who like to be adventurous!
_Also the settings format is different._

Probably we will have a new Beta soon with some improvements.

**Reply by Ciutnik** [reply]
> What do you mean with the "reset to default", when this happen?

The program started normally, but the profile change resulted in FC. After restarting the app, all settings are gone (font sizes, colors, even the previously selected router, ...)

Edit: Even worse, the option to restore preferences disappeared from the menu 🙂

**Reply by devemux86** [reply]
FC?

> the option to restore preferences disappeared from the menu

Do you mean the Import in the Settings?

Import settings from file is available with long press in the profile manager.

> Profile selection / import / export in settings are removed

See also here:
- https://github.com/devemux86/cruiser/discussions/165#discussioncomment-6800089

**Reply by Ciutnik** [reply]
Oh yeah, I missed managing preferences from profiles screen. But you can't import a .pref file anyway.

**Reply by devemux86** [reply]
Please read carefully the announcement:
- https://github.com/devemux86/cruiser/discussions/165#discussioncomment-6800089

There is a new format and when app first starts, it already has the last settings.

**Reply by Ciutnik** [reply]
> when app first starts, it already has the last settings.

It was like this for a while, until the unexpected forced closure of the app... The idea of ​​temporarily restoring the .pref import option seems like a good one 👍

**Reply by airald** [reply]
My settings are not restored to default when trying beta (2)

**Reply by airald** [reply]
I correct myself: after install Beta 3 (3.0.31-3), Cruiser stops when I change profile, and also all settings are lost on first start. Then I restored easily (I always have different backups from my preferences that I store on the cloud to be able to use in my different devices).

**Reply by devemux86** [reply]
@airald No automatic restart? What Android / device?

There is an improvement we need in the storage.

**Reply by devemux86** [reply]
@airald 

When you change the app language and exit Settings to return to the map,
doesn't the app restart automatically with the new language?

**Comment by devemux86** [comment]
We could have (at least during the Beta) the Import .pref in Settings to help the testing.

I will make a new Beta apk soon.

**Comment by ramondevic** [comment]
I saw the new option that supports Catalan, I installed it and lost all the particular color settings (and I think the favorites file too) and I couldn't find the option to import the preferences I had saved, so I uninstalled and reinstalled the Google Play version and re-imported my preferences from the .pref file.
However, I have seen that from now on the exports/imports work differently, I think it should be possible to maintain the old preferences formats ... otherwise they are useless.

Maybe I was too quick to install a beta version, I'll be more careful from now on :(

**Comment by ramondevic** [comment]
In the few minutes I've had the Catalan version installed I've seen some translation errors that look like writing, I'll go through the list later when I install it again.
It also seemed to me that the "search" option did not work.

**Reply by airald** [reply]
I've also found my own errors, they are also sent to Emux

**Comment by devemux86** [comment]
Cruiser 3.0.31-3 (**Beta**) is available with the .pref import option (in Settings).

There is an issue with the Android storage of the profiles,
I'm looking for it to be fixed in an upcoming Beta version.

**Reply by WolfgangZ42** [reply]
Because I start today for an 8-day motorcycle tour, I deliberately did not install 3.0.31-2/3. 
I am already since the beginning of my Cruiser use beta tester and assumed that already the 3.0.31 is a beta version - or this has the general available status? And I don't want to leave the beta program now, because I might have to install another version.
Isn't 3.0.31-3 rather an alpha test version? Normally I would install it - but not now, just before my tour.

**Reply by Ciutnik** [reply]
In my humble opinion, rather stay with 3.0.31 from the Store. It is stable as a rock. As @devemux86 wrote, the above published versions may have problems with profiles, and in my case it was force closed several times. In terms of everyday use (if you don't care about Polish or Catalan 😉), you probably won't notice the difference.

Well, I wish you a warm, dry and **safe** tour, full of beautiful curves and views. Good luck 👍

**Reply by devemux86** [reply]
@WolfgangZ42 you mean you are a member of Cruiser (Beta) on Google Play:
- https://github.com/devemux86/cruiser/discussions/36

Cruiser 3.0.31 is the latest public release.

---

The apk here is offered separately because it needs thorough testing by users who know how to install the apk. :slightly_smiling_face:

**Comment by DolomitesTom** [comment]
When is it available via Google Play Store?

**Reply by devemux86** [reply]
@DolomitesTom we will see, when things are stable.
Now it is time to improve and test the profile manager.

**Comment by devemux86** [comment]
I forgot to mention that changing profiles (with OK) restarts the app.

(maybe you meant this by force close)

**Reply by Ciutnik** [reply]
> maybe you meant this by force close

After each profile change, the application closes without warning, but does not start itself. You have to run it manually, already with a changed active profile. I suppose, it would be good to have the message "Application restart required" and then "OK / Cancel".

**Reply by devemux86** [reply]
That's strange, what Android / device?
App restart is the same as when we change some basic option in Settings, e.g. the app language.

Indeed, we need a message here to explain it to users.

**Reply by Ciutnik** [reply]
> what Android / device?

This happens on a Xiaomi phone (Android 10) and on a Samsung tablet (Android 13).

**Reply by 0709wiwiwi** [reply]
Android 10 Xiaomi Mi A2 Lite (Android one)

v 3.0.31-2
The app closes & does not autorestart.
In fact I do not notice profile changes.
Internal maps & themes not memorised.
External maps & themes not possible.

v 3.0.31-3
Dropbox > public > This Folder is empty.

**Reply by devemux86** [reply]
There are a few things currently held in common across profiles:

- Storage folder (downloads)
- Loaded maps / poi
- Map position

This is to avoid confusion when changing profiles.
At least the storage folder is important. About the other things?

> Dropbox > public > This Folder is empty.

We need some improvements before we continue the Beta again.

**Reply by devemux86** [reply]
Those of you having issues with app restart, doesn't it work all this time when changing some settings?

e.g. change app language, exit settings, doesn't the app restart automatically with the new language?

**Reply by Ciutnik** [reply]
I also did tests that changing the language from Polish to English, unfortunately it does not help. But interestingly, profile names are always in Polish, even when the entire interface is English. In addition, in Xiaomi (not in Samsung), the order of profiles on the list changes after each application closure.

Edit: An error report has been generated on Xiaomi:

`java.lang.ClassCastException:
java.util.ArrayList cannot be cast to java.util.Set
	at com.devemux86.profile.e.i(SourceFile:97)
	at com.devemux86.profile.f$d.run(SourceFile:45)
	at java.lang.Thread.run(Thread.java:919)`

**Reply by devemux86** [reply]
And here with native Android everything works perfectly (that's why the Beta was published).
There are always differences in the Android devices that make everything more complicated.

> An error report has been generated on Xiaomi

@Ciutnik thanks, I will fix it. :+1:

**Comment by devemux86** [comment]
The storage issue should be solved, Android likes to play games. :joystick: 

Let's see why restart behaves differently on some devices.

**Reply by Ciutnik** [reply]
> Android likes to play games

Too bad it's an endless game of Google, against OEMs, against users. Fortunately, the game seems to be fairly even 😉

**Reply by devemux86** [reply]
The annoying thing is that developers waste valuable time to solve such issues,
instead of developing new features. :slightly_smiling_face:

**Comment by devemux86** [comment]
@airald @Ciutnik @0709wiwiwi 

Please check your devices and let me know:

- Go to "Settings | Application"
- Change the app language
- Exit settings (press back) to return to map
- Does the app restart automatically with the new language?

**Reply by Ciutnik** [reply]
Yes, there is nothing wrong with this workflow. The language changes and the application continues to function normally. Only changing the profile kills the application.

**Reply by devemux86** [reply]
@Ciutnik When you exit settings and return to the map, the **app restarts automatically** with the new language?

> Only changing the profile kills the application.

This is actually the same call to restart the app, but it doesn't seem to work everywhere in this case.

**Reply by Ciutnik** [reply]
> When you exit settings and return to the map, the app restarts automatically with the new language?

Yes. It takes a fraction of a second. I changed the language between English, German and Polish many times and everything works fine. Of course the language is remembered. What's more, I changed the theme from dark to light and back and it works too.

**Reply by Ciutnik** [reply]
That Java bug report that I quoted above occurs moments after the application is killed by a profile change. If you could get rid of it, maybe the problem wouldn't exist anymore.

**Reply by devemux86** [reply]
> That Java bug report that I quoted above occurs moments after the application is killed by a profile change. If you could get rid of it, maybe the problem wouldn't exist anymore.

This should be fixed in Cruiser 3.0.31-4 (Beta).
@Ciutnik can you test again?

**Comment by devemux86** [comment]
Cruiser 3.0.31-4 (**Beta**) is available with the storage and error fix.
Also with other improvements and translation updates.

More improvements are planned, for now please test this intermediate Beta version.
e.g. see when switching profiles if the automatic app restart works.

**Reply by Ciutnik** [reply]
Good job! 👍
Thanks for the quick fix. Now changing the profile (language, theme) does not kill the application. 

Profile names do not change after changing the language of the application. They are always Polish for me (because the language of the system?), and in addition, in Xiaomi (Android 10) the order of profiles is not remembered.

**Reply by devemux86** [reply]
> Now changing the profile (language, theme) does not kill the application.

So the cause was the exception in your report. :+1: 

> Profile names do not change after changing the language of the application.

The base profiles (with the star) should have their original translated names.
We need a change here to fix this.

New created profiles from the users can have any custom name.

> in Xiaomi (Android 10) the order of profiles is not remembered

You reorder them, press OK and when app restarts they have the old order?
Or when you re-open the profile manager without restarting the app?

**Reply by Ciutnik** [reply]
> in Xiaomi (Android 10) the order of profiles is not remembered

After each restart of the application, the order of the profiles is reset. No matter if I normally exit the app and restart, or if I change language and it restarts automatically. Without restarting the application, re-opening the profile manager does not change the order. And what's interesting, after each launch of the application, this order is different! Even if I don't change anything in the settings.

In Samsung with Android 13, this does not occur.

**Reply by devemux86** [reply]
When their order is different, when you switch profiles, are their settings correct?

> In Samsung with Android 13, this does not occur.

!!!

**Reply by Ciutnik** [reply]
> When their order is different

Each restart of the application causes a random (?) order of profiles. 

Look, the first picture is the order manually set and confirmed with OK. Until I leave the app it stays like this all the time. Each next picture is the profile manager after restarting the application. Weird...

<img src="https://github.com/devemux86/cruiser/assets/122647782/d9c2806c-c9d4-4872-9b2f-ce995c401b6a" width="24%"/>

<img src="https://github.com/devemux86/cruiser/assets/122647782/7dbdb368-1f33-4c33-8315-ae4ac31b6634" width="24%"/>

<img src="https://github.com/devemux86/cruiser/assets/122647782/35bcf822-2cf6-4354-9af6-9123c408a9ce" width="24%"/>

<img src="https://github.com/devemux86/cruiser/assets/122647782/fe25991d-2485-45ae-8e8a-92f635b5108b" width="24%"/>

**Reply by devemux86** [reply]
@Ciutnik can you create 1-2 new profiles (based on any default profile) and see if this happens again?

**Reply by Ciutnik** [reply]
I've created an additional two profiles (Motorcycle 2, Car 2) and they behave exactly the same as the default profiles. They randomly mix with the others after each restart of the application.

**Reply by Ciutnik** [reply]
I installed 3.0.31-4 on another Samsung (Android 12), there is no mixing of profiles here either. The conclusion is that Xiaomi is mixing 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks for the tests! I will check it.

Can you also see when you change the active profile if its settings are correct?

> The conclusion is that Xiaomi is mixing

Android world never stops surprising us!

**Reply by Ciutnik** [reply]
> Can you also see when you change the active profile if its settings are correct?

Yes, it seems that preferences are remembered. So on quickly, I changed only the color (accent) of the application in the profiles.

**Reply by Ciutnik** [reply]
Just a note: when installing a new version of the application containing a new profile manager, only the active profile takes the preferences (colors, sizes) of the older version of the application? All other profiles assume some default values. It would probably be best if all profiles adopted the prefs of the older version of the application.

**Reply by devemux86** [reply]
@Ciutnik thanks for the note!

This is something I also wanted to discuss with all of you.

> when installing a new version of the application containing a new profile manager, only the active profile takes the preferences (colors, sizes) of the older version of the application? All other profiles assume some default values

Currently, this is how it works.

> It would probably be best if all profiles adopted the prefs of the older version of the application.

- When we switch to a profile that has never been used before, you mean it should use the existing app settings?
- When we create a new profile, we select a base profile to inherit its settings
  - We can reset all profiles to their default settings at any time

**Reply by Ciutnik** [reply]
> When we switch to a profile that has never been used before, you mean it should use the existing app settings?

For brand new users this doesn't matter, but for those upgrading from an older version, when the settings were common to all profiles, this behavior is expected. And then let them adjust what they want to a specific profile. 

But now I think it is only a matter of a few minutes spent on re-setting the profile ONCE. If the implementation were to take you longer, maybe it's a waste of trouble? Especially since you can import settings from another profile!

> When we create a new profile, we select a base profile to inherit its settings

This is a very good solution.

**Reply by devemux86** [reply]
> It would probably be best if all profiles adopted the prefs of the older version of the application.

> The base profiles (with the star) should have their original translated names.

Both will be available in the next Beta version.

**Comment by devemux86** [comment]
Cruiser 3.0.31-5 (Beta) is available with various improvements.

@Ciutnik @0709wiwiwi 
Please check on your Xiaomi if the profiles order after restart is correct.

**Reply by Ciutnik** [reply]
Bad news this time... The profiles are still mixed up 😟

**Reply by devemux86** [reply]
How these devices use Android?! :upside_down_face:

**Reply by Ciutnik** [reply]
> How these devices use Android?! 🙃

In an amazingly Chinese way 😉

**Reply by airald** [reply]
I use a Redmi Note 11 Pro 5G (MIUI 14). Profiles order are correct after restart (maybe Redmi is not exactly the same as Xiaomi, but I think that they share a lot of resources). Profile names are in the correct language.

**Reply by Ciutnik** [reply]
I expect that MIUI 12 may cause trouble. Chinese overlays greatly interfere with Android. I once turned off "MIUI Optimization" - maybe that causes additional problems? Or anything else 🙂The phone is old, the system is old, I don't think it's worth fighting with.

**Reply by airald** [reply]
MIUI Optimization can cause problems when app is running on foreground, I don't have problems with this feature when running apps "normally".

**Reply by devemux86** [reply]
@Ciutnik 

Have you seen anything similar in the favorites?
(in the 2 lists of the favorite groups or favorites)

**Reply by Ciutnik** [reply]
Indeed yes! I hadn't noticed it before! The order of the Favorite groups is similarly mixed up, but within one group the points do not mix - their order is kept here.

One more observation: the last opened group of Favorites is not remembered. Each restart of the application causes a different group to be opened.

**Reply by 0709wiwiwi** [reply]
A10 Xiaomi MiA2 Lite through Android one which promises stock rom experience.  The profile order keeps changing. Apart from curious, this may (more tests still needed) possibly not cause any inconveniences ?

>Have you seen anything similar in the favorites?

Favourites I myself do not use (yet).

**Reply by devemux86** [reply]
> the last opened group of Favorites is not remembered

This was already implemented after @DolomitesTom request: https://github.com/devemux86/cruiser/discussions/45#discussioncomment-6454899
But if the device mixes the favorite groups, then it is expected to display a random one.

It is strange that these devices exhibit such unusual behavior when all the others are working correctly. :thinking: 

I have some ideas on how to deal with this, we'll see.

**Reply by Ciutnik** [reply]
@0709wiwiwi 
> Android one which promises stock rom experience

Was the promise not kept? 🙂

**Reply by 0709wiwiwi** [reply]
Well, so far I haven't seen "TikTok" to appear anyway.

**Comment by devemux86** [comment]
Cruiser 3.0.31-6 (Beta) is available with various improvements.

@Ciutnik @0709wiwiwi
Please check on your Xiaomi if the profiles order after restart is correct.
Better to reorder the profiles slightly, press OK and then restart the app.

**Reply by Ciutnik** [reply]
Bingo! 🎖️ Congratulations! The order of the profile is already remembered! 👍 But there are still mixed favorites.

> Better to reorder the profiles slightly, press OK and then restart the app.

I did that from the very beginning.

**Reply by devemux86** [reply]
I can't believe that we have to do such workarounds to fix every OEM Android problem.
Luckily it wasn't too difficult. :slightly_smiling_face:

> But there are still mixed favorites.

I only applied the workaround to the profiles for testing.
Since it works, I will also apply it to the favorite groups.

**Reply by 0709wiwiwi** [reply]
>The order of the profile is already remembered! 

👍.

**Comment by devemux86** [comment]
Cruiser 3.0.31-7 (Beta) is available with various improvements.

@Ciutnik favorites order is implemented in this version.

**Reply by Ciutnik** [reply]
Thank you very much! I can confirm proper functioning 👍 Even in Xiaomi 😀

**Comment by airald** [comment]
Altitude is shown even if not in navigation or follow mode, interesting!

**Reply by devemux86** [reply]
@airald please also check the related discussion here:
- https://github.com/devemux86/cruiser/discussions/287#discussioncomment-6767023

**Reply by airald** [reply]
Thank's. Usually I see unread topics in Bold, but not sometimes, so I miss some posts...

**Comment by devemux86** [comment]
Cruiser 3.0.31-8 (Beta) is available with various improvements.

**Comment by devemux86** [comment]
Cruiser 3.0.31-9 (Beta) is available with various improvements.

**Reply by Ciutnik** [reply]
The new profile arrangements are great! Only the settings for the current profile are shown, without causing confusion. Thanks!

But file storage is set separately for each profile. Shouldn't this be an application setting common to all profiles? Like, for example, the application language.

**Reply by devemux86** [reply]
> But file storage is set separately for each profile. Shouldn't this be an application setting common to all profiles?

@Ciutnik are you sure , how did test it?
The storage folder is global.

> Like, for example, the application language.

Storage folder and loaded maps / poi are global to avoid confusion.
We could make the app language and any other setting global too.

**Reply by Ciutnik** [reply]
> are you sure , how did test it?

I just changed the profile to a previously unused one. An online map appeared, and a Downloads / Map was empty. I checked the storage directory and it was /Android/data. I usually have /Android/media. But after changing the storage settings to /media everything is OK. Wanting to go back to the previous folder /data again, the expected question about deleting data appeared, so I did not continue trying...

But this may not be a valid example. Previously, I loaded the .pref file into different profiles, reset the settings and did terrible things with the application many times 😀

**Reply by Ciutnik** [reply]
> We could make the app language and any other setting global too.

I think the language should be global. Perhaps the theme of the app as well. These preferences are rather constant for each user.

**Reply by devemux86** [reply]
@Ciutnik thanks for the details.

Can you check the storage folder in the "previously unused one" profile? Do you have it as json?
Perhaps is was a folder that cannot be read anymore, so the default Android/data was used.

It should use the current folder, so we need a code improvement here. :slightly_smiling_face: 

---

App language could be global.

I am not so sure about the app theme.
There may be cases with profiles for day / low light use which involves app theme + map colors.

**Reply by airald** [reply]
I just updated to -9 and when switching to any unused profile (I think...) the map config remains ok. But I experienced It on some previous versions. Just change the storage folder and everything is ok again.

**Reply by airald** [reply]
Tested on another phone with one real unused profile. When changing to this profile, the map switches to online map, but when returning to used profile, the map type remains online.

The storage folder is global, does not change. 

I think that profile can switch to online but can NOT switch to downloaded maps.

**Reply by devemux86** [reply]
> Tested on another phone with one real unused profile.

Remember that all base profiles are filled with app settings,
one time at the first app start after the profile introduction.

The expected process currently should be that when we switch profiles,
the storage folder and the active maps / poi remain unchanged.

If something else happens, it should be some error.

**Reply by devemux86** [reply]
>  Tested on another phone with one real unused profile.

> Remember that all base profiles are filled with app settings,
one time at the first app start after the profile introduction.

This was implemented later, _not in the first Beta_,
so it depends what Beta you first installed in that phone.

**Reply by devemux86** [reply]
If you can, it is best to make the tests with new installs of Cruiser.
Install Cruiser from Google Play and then install the latest Beta.

**Reply by Ciutnik** [reply]
@devemux86 
> Can you check the storage folder in the "previously unused one" profile?

I guess it's been too many clicks since the last test for me to be able to reproduce anything. Now everything is OK, but as I recall, I moved the .prefs file between devices. Maybe the directory paths were different and that's why the profile defaulted. 

Overall great job! 👍

**Comment by devemux86** [comment]
Cruiser 3.0.31-10 (Beta) is available with various improvements.

**Reply by Ciutnik** [reply]
Beta v10. Something strange. I cannot set a app language other than the one previously selected (Polish). The app resets but the language doesn't remember.

**Reply by devemux86** [reply]
@Ciutnik thanks for the report!

I can reproduce it, it will be fixed in the next update.

**Comment by devemux86** [comment]
Cruiser 3.0.31-11 (Beta) is available with various improvements.

**Comment by WolfgangZ42** [comment]
Back from an 8-day gravel tour in Friuli with 7 motorcycles (navigation: 3 Cruiser, 2 Garmin, 1 TomTom). Guess by which tool the tours were planned and guided. Totally offline (often no network available, one Cruiser cell phone even without SIMcard), planned with mountainbike profile, ridden with motorcycle profile to be properly routed to the respective starting point. Various rescheduling, even reversal of routes, all perfectly solved offline. TomTom now has one less customer. **Thanks Emux we can use the best navigation system!**
And in the next days I'll have a look at 3.0.31-11 ...

**Reply by devemux86** [reply]
@WolfgangZ42 many thanks!

> planned with mountainbike profile, ridden with motorcycle profile to be properly routed to the respective starting point

Can you explain more, why using the mountain bike profile?
Does the cruiser-driving profile need any improvements?

**Reply by WolfgangZ42** [reply]
When planning gravel roads, I had the impression that with the profile "Mountainbike" given tracks can be converted more easily into routes, e.g. through fewer shaping points. This does not apply to all tracks and depends of course on the OSM definitions. Sometimes it's even impossible to load a track with "Motorcycle":
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/e242ef06-bc0b-44c0-8437-140ca7cf036d"/> but it works with "Mountainbike" - if you'll try it: [Friaul_0830.gpx.txt](https://github.com/devemux86/cruiser/files/12516305/Friaul_0830.gpx.txt)

If I then drive this routes (exported as navigation gpx) with the profile "Motorcycle", I get fewer surprises in the admittedly rare reschedulings or the way to the starting point of the gravel route. 

> Does the cruiser-driving profile need any improvements?

I think not - it depends really of the OSM definition.

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the track!

I examined it using the internal [driving-cruiser](https://github.com/devemux86/cruiser/discussions/272) BRouter profile:
- With default settings we get the message "Point 30 is too far from a route".
The way in this area is: https://www.openstreetmap.org/way/232280904 with **tracktype=grade4**
So we need to enable the option "**Use gravel roads**".

- Then we get the message "Point 27 is too far from a route".
The way in this area is: https://www.openstreetmap.org/way/34647801 with **motor_vehicle=no**
Did you drive this road?

Or do we need some option for off-road users?

**Reply by WolfgangZ42** [reply]
> So we need to enable the option "Use gravel roads".

That is always enabled in my "Motorcycle" profile

> Did you drive this road?

Sure we did - may be we overlooked some traffic signs? I can't remember, otherwise I'd edit the OSM data.

> Or do we need some option for off-road users?

I think that's not necessary ... but I'll keep it in mind. 

My post was just an answer of your questions - not a request. With the mix of "Mountainbike" and "Motorcycle" it worked for me. 
And now the switch by profile manager between them is even faster - good job!

**Reply by devemux86** [reply]
@WolfgangZ42

If your friends have suggestions for Cruiser improvements or how to make it easier,
please write them so we can discuss them! :slightly_smiling_face:

I understand that the BRouter app installation is an extra step for everyone.
Also the nearby POI need download of maps / poi and enable in settings.

I'm still thinking of ways to make Cruiser operation simpler for new users.

**Reply by WolfgangZ42** [reply]
> I understand that the BRouter app installation is an extra step for everyone.

As long as it is just an installation and nothing more (for BRouter (offline) service) it should be no problem.

> Also the nearby POI need download of maps / poi and enable in settings.

Downloads are self-explanatory IMHO - settings are becoming complicated inbetween. 

I've already written something for "newcomers": 
[Idiots_Guide_en.pdf](https://github.com/devemux86/cruiser/files/12523349/Idiots_Guide_en.pdf)
Feel free to share it with anybody. 
The German Version I gave to some of my friends. Now I've to update it (.json instead of .pref) ...

**Reply by 1maxi2** [reply]
Idiots scheint für mich sehr interessant.  Wie kann ich die Datei auf deutsch übersetzen lassen ?

**Comment by devemux86** [comment]
Cruiser 3.0.32-b12 is available.

**Comment by devemux86** [comment]
Cruiser 3.0.32-b13 is available.

**Reply by Ciutnik** [reply]
I noticed a problem using the on-screen zoom keys right after starting navigation, before I move. Hardware (volume) and double click and swipe work fine.

The magnification changes slightly for a fraction of a second and immediately returns to the previous one. ~~This is happening on two Samsung tablets. Zoom works fine on Xiaomi phone.~~

**Reply by devemux86** [reply]
@Ciutnik can you mention the steps you used?
Start navigation and immediately press zoom buttons?

Is this because of the auto zoom?
Auto zoom may not happen immediately.

What zoom did it return?

**Reply by Ciutnik** [reply]
I select a point, set it as a destination and start navigation. The on-screen keys no longer work. Either immediately or after a few clicks in +/-. Maybe the distance of about 50 m from the nearest road matters? The zoom is set to the maximum defined in the settings (17).

Auto zoom set to Distance, or Speed + Distance. If it's just Speed it doesn't happen.

Also on Xiaomi I reproduced this behavior.

**Reply by devemux86** [reply]
It seems the zoom buttons do not disable temporarily the auto zoom, like the other zoom gestures.
So it immediately auto zooms according to the selected settings.

**Reply by devemux86** [reply]
Nothing has changed in this area, so we had such workflow all this time?
We need to fix it. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> It seems the zoom buttons do not disable temporarily the auto zoom,

Yes, that's what it looks like. Maybe it's always been like this? I very rarely use the onscreen zoom buttons. On a motorcycle always hardware, and at home double-click + swipe.

**Reply by devemux86** [reply]
> It seems the zoom buttons do not disable temporarily the auto zoom

It will be fixed in the next Cruiser update.

**Comment by devemux86** [comment]
For reference all zoom gestures in Cruiser:

- **Two finger zoom**: use two fingers and pinch to zoom in / out
- **Double tap with one finger** to zoom in
- **Tap with two fingers** to zoom out
- **One finger zoom**: double tap with one finger, keep finger on screen, move finger to zoom in / out
- **Use volume keys** to zoom in / out (enable in map settings)

**Comment by devemux86** [comment]
Cruiser 3.0.32-b14 is available.

If you don't see anything that needs fixing, especially in the new **profile manager**,
I can do a release in the next few days.

**Comment by devemux86** [comment]
Cruiser 3.0.32-b15 is available with improvements and:
- https://github.com/devemux86/cruiser/discussions/305

**Comment by devemux86** [comment]
Cruiser 3.0.32-b16 is available.

**Reply by Ciutnik** [reply]
The new profile menu works great on the phone! Unfortunately, the menu does not appear on tablets (8" and 10").

**Reply by devemux86** [reply]
@Ciutnik after tapping the right icon the menu does not appear?

Do you tap on the right icon (not left or right area)?
If you tap outside the profiles is activated. Try it on a non active profile's row.

What tablets / Android?

**Reply by Ciutnik** [reply]
I tried everything. Touching the center or icon activates the profile. This is ok. However, a short touch on the "three dots" highlights the line for a while (as if it were a long touch while dragging), but the menu does not appear.
Both Samsung tablets, 10" Android 12 and armored 8" Android 13.

EDIT: It looks like a short press will immediately activate drag mode before the menu shows up.

**Reply by devemux86** [reply]
> Touching the center or icon activates the profile.

@Ciutnik do you mean the left icon?

Can you reorder the profiles with the right icon?

---

We can keep the menu _also_ via long press. It can be useful in large screens instead of going right for the menu icons.

**Reply by devemux86** [reply]
> EDIT: It looks like a short press will immediately activate drag mode before the menu shows up.

So it seems something works differently on Samsung devices? Probably a system timer.
Then we cannot have both actions on the same icon.

We can either move the drag back to the left icon.
Or have the drag working with the long press (which now is free).

**Reply by Ciutnik** [reply]
> Can you reorder the profiles with the right icon?

Yes, everything works. Tapping the left icon does nothing. I recorded the screen, but I'm wondering how to quickly share 11 MB with you... Maybe by e-mail if you want?

**Reply by Ciutnik** [reply]
> Or have the drag working with the long press (which now is free).

Definitely yes! This is a common way to activate drag! We don't do this very often.

**Reply by Ciutnik** [reply]
Haha, by the way, I have the opportunity to try out Samsung's quick share feature, which I didn't need to use before. This is the recorded screen:

https://quickshare.samsungcloud.com/w4bzq4PmuYau

Edit: It just occurred to me that this share might be only for Samsung users? If so, I'm sorry. I can use something "normal" 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks the video is visible!

Our possible alternatives seem to be:
- Menu on right icon, drag on long press
- Menu on both right icon + long press, drag on left icon

In any case I need to make more tests.

**Reply by airald** [reply]
On my Samsung tab everything is ok except menú, never appears.

**Reply by devemux86** [reply]
It seems Samsung uses smaller timer in these events so they cannot work together.

We need to change somehow the implementation (@airald see [above](https://github.com/devemux86/cruiser/discussions/299#discussioncomment-6950621)).

**Reply by Ciutnik** [reply]
> Or have the drag working with the long press (which now is free).

Do you mean the right icon (six dots) or the area beyond that icon? Initially, I understood that it was possible to either press it briefly for menus or long press it for dragging.
Now I think the right icon cannot be short/long pressed due to a quirk of Samsung, and therefore 

> Menu on both right icon + long press, drag on left icon

seems optimal. To make it easier to show menu (in two ways) rather than rarely used dragging.

**Comment by devemux86** [comment]
Cruiser 3.0.32-b17 is available.

**Reply by airald** [reply]
Profiles menu working properly on Xiaomi phone and Samsung Tab. Select, move, delete and menu.

**Comment by devemux86** [comment]
Cruiser 3.0.32-b18 is available with the new lists also in favorites / waypoints.

**Reply by airald** [reply]
The useability of favorites is now much better, more intuitive and easy to enter different groups and specially is not easy to enable or disable visibility and navigation alerts accidentally. Working properly in Samsung tab too.
